src/services/project/project-router.ts

```diff
@@ -0,0 +1,170 @@
+                schema: NoTeamFoundErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const ownerId = req.params.ownerId;
+        const projectDetails = await Models.ProjectInfo.findOne({ ownerId });
+
+        // Return no project found if no details were found
+        if (!projectDetails) {
+            return res.status(StatusCode.ClientErrorConflict).send(NoTeamFoundError);
+        }
+
+        return res.status(StatusCode.SuccessOK).send(projectDetails);
+    },
+);
+
+// GET details of user's team
+projectRouter.get(
+    "/details",
+    specification({
+        method: "get",
+        path: "/project/details/",
+        tag: Tag.PROJECT,
+        role: null,
+        summary: "get details of user's team",
+        body: ProjectProjectsSchema,
+        responses: {
+            [StatusCode.SuccessOK]: {
+                description: "The user's team",
+                schema: ProjectProjectsSchema,
+            },
+            [StatusCode.ClientErrorConflict]: {
+                description: "No team found for the user",
+                schema: NoTeamFoundErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const { id: userId } = getAuthenticatedUser(req);
+        const userMapping = await Models.ProjectMapping.findOne({ userId });
+        if (!userMapping) {
+            return res.status(StatusCode.ClientErrorNotFound).send(NoTeamFoundError);
+        }
+        const ownerId = userMapping?.teamOwnerId;
+
+        // find project associated with the ownerId
+        const projectDetails = await Models.ProjectInfo.findOne({ ownerId: ownerId });
+
+        return res.status(StatusCode.SuccessOK).send(projectDetails?.toObject());
+    },
+);
+
+// POST create project/team
+projectRouter.post(
+    "/create",
+    specification({
+        method: "post",
+        path: "/project/create/",
+        tag: Tag.PROJECT,
+        role: null,
+        summary: "create a new project/team",
+        body: CreateProjectRequestSchema,
+        responses: {
+            [StatusCode.SuccessOK]: {
+                description: "The new team",
+                schema: ProjectProjectsSchema,
+            },
+            [StatusCode.ClientErrorConflict]: {
+                description: "The user already has a team",
+                schema: UserAlreadyHasTeamErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const { id: userId } = getAuthenticatedUser(req);
+        const details = req.body;
+
+        const project: Project = {
+            ownerId: userId,
+            ...details,
+        };
+
+        // ensure teamOwner hasn't already made a team
+        const ownerExists = (await Models.ProjectInfo.findOne({ ownerId: userId })) ?? false;
+        if (ownerExists) {
+            return res.status(StatusCode.ClientErrorConflict).send(UserAlreadyHasTeamError);
+        }
+
+        const newProject = await Models.ProjectInfo.create(project);
+
```

Comment:
We'd probably also have logic for adding a projectMappings of the user to themselves, since they should be part of their own group.